### ISSUE TITLE
Remove 'with' usage does not work if the variable is a parameter for the procedure

### ISSUE THREAD

**Issue by FloOM25** [anzwdev/al-code-outline#372]
It removes the "with" statement but not adds the var to the code. 
Just seems to happen when the var is used for parameters.

![issue before action](https://user-images.githubusercontent.com/108056115/175250879-5d5138ca-fa2a-4fac-ba43-d9eb71c7628e.png)
![issue after action](https://user-images.githubusercontent.com/108056115/175250875-41fb95a0-8bff-40b6-8b02-23f3344c85dd.png)

**Comment by Splace13** [comment]
The same thing seems to happen with global variables. The "with" statement gets removed but the variable name is not added to the code.

**Comment by anzwdev** [comment]
I was trying to replicate the issue, but it works for me without any problems. Is it happening with all "with" blocks, or only in a single, special case? Is it broken in inside all object types or just a single one? Are you using the latest Microsoft "AL Language" extension?

**Comment by FloOM25** [comment]
I did another try in a new, clean, project.
You´re right. It works like intended there. 

It seems to happen in larger workspaces with more files.
But not sure if it´s a case for u then.

**Comment by frozenbrain** [comment]
I was able to reproduce this issue by changing the package cache path in my .vscode/settings.json:
```
{
    "al.packageCachePath": "foo"
}
```

I then deleted the old .alpackages directory, re-downloaded symbols and tried removing the with-statement in the following procedure:
```
procedure Foo(SalesHeader: Record "Sales Header")
begin
    with SalesHeader do begin
        Message("Bill-to Name");
    end;
end;
```
Which resulted in the following (broken) code:
```
procedure Foo(SalesHeader: Record "Sales Header")
begin
    Message("Bill-to Name");
end;
```

It seems like the al.packageCachePath setting is completely ignored. It works as long as there are symbol packages in the default .alpackages directory. Even just deleting the package cache without touching the setting causes the command to silently fail.
